src/pe/dynreloc.rs

```diff
@@ -0,0 +1,1258 @@
+    block_rva: u32,
+}
+
+impl Iterator for DynRelocRelocationIterator<'_> {
+    type Item = error::Result<DynRelocRelocation>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset >= self.bytes.len() {
+            return None;
+        }
+
+        let reloc = match self.symbol {
+            IMAGE_DYNAMIC_RELOCATION_ARM64X => {
+                let record = DVRTArm64XFixupRecord(
+                    self.bytes
+                        .gread_with::<u16>(&mut self.offset, scroll::LE)
+                        .ok()?,
+                );
+
+                if record.is_null() {
+                    return None; // Reached the last entry
+                }
+
+                DynRelocRelocation::Arm64X(DynRelocArm64X {
+                    symbol: self.symbol,
+                    block_rva: self.block_rva,
+                    record,
+                })
+            }
+            IMAGE_DYNAMIC_RELOCATION_GUARD_IMPORT_CONTROL_TRANSFER => {
+                let record = ImportControlTransferDynReloc(
+                    self.bytes
+                        .gread_with::<u32>(&mut self.offset, scroll::LE)
+                        .ok()?,
+                );
+
+                DynRelocRelocation::Import(DynRelocImport {
+                    symbol: self.symbol,
+                    block_rva: self.block_rva,
+                    record,
+                })
+            }
+            IMAGE_DYNAMIC_RELOCATION_GUARD_INDIR_CONTROL_TRANSFER => {
+                let record = IndirectControlTransferDynReloc(
+                    self.bytes
+                        .gread_with::<u16>(&mut self.offset, scroll::LE)
+                        .ok()?,
+                );
+                let reloc = Into::<RelocationWord>::into(record);
+
+                if record.page_relative_offset() == 0 || reloc.reloc_type() == 0 {
+                    return None; // Reached the last entry
+                }
+
+                DynRelocRelocation::Indirect(DynRelocIndirect {
+                    symbol: self.symbol,
+                    block_rva: self.block_rva,
+                    record,
+                })
+            }
+            IMAGE_DYNAMIC_RELOCATION_GUARD_SWITCHTABLE_BRANCH => {
+                let record = SwitchableBranchDynReloc(
+                    self.bytes
+                        .gread_with::<u16>(&mut self.offset, scroll::LE)
+                        .ok()?,
+                );
+
+                if record.page_relative_offset() == 0 {
+                    return None; // Reached the last entry.
+                }
+
+                DynRelocRelocation::Branch(DynRelocBranch {
+                    symbol: self.symbol,
+                    block_rva: self.block_rva,
+                    record,
+                })
+            }
+            IMAGE_DYNAMIC_RELOCATION_FUNCTION_OVERRIDE => {
+                // This is unreachable. See impl Iterator for DynRelocBlockIterator.
+                unimplemented!()
```

Comment:
you're 100% sure this is not reachable? To be clear, I mean even with malformed, fuzzer generated binaries etc., this should truly be impossible to reach.